src/services/codefixes/importFixes.ts

```diff
@@ -644,6 +665,19 @@ namespace ts.codefix {
             Debug.fail("Either the symbol or the JSX namespace should be a UMD global if we got here");
         }
 
+        const { module, allowSyntheticDefaultImports } = context.compilerOptions;
+
+        // Prefer to import as a synthetic `default` if available.
+        if (allowSyntheticDefaultImports || module === ModuleKind.System && allowSyntheticDefaultImports !== false) {

```

Comment:
Add a function to `src\core.ts` to `getAllowSyntheticDefaultImports(compilerOptions)` that has this logic, and share it with the checker as well.